examples/Tests/ParametricEq.hpp

```diff
@@ -0,0 +1,584 @@
+#pragma once
+
+#include <vector>
+#include <string>
+#include <DspFilters/Filter.h>
+#include <DspFilters/Butterworth.h>
+#include <DspFilters/ChebyshevI.h>
+#include <iostream>
+
+namespace examples
+{
+
+struct ParametricEq
+{
+    static constexpr auto name() { return "Parametric Equalizer"; }
+    static constexpr auto c_name() { return "ngry_parameq"; }
+    static constexpr auto uuid() { return "ec972d12-edba-4a9c-9736-9176efad17e3"; }
+
+    // ********************************************************************************
+    //                              Structural Part of the object
+    // ********************************************************************************
+    struct ins
+    {
+        struct 
+        {
+            static constexpr auto name() { return "Input"; }
+            const double** samples;
+            static constexpr auto channels() {  return 2; }
+            float sampleRate;
+        } audio;
+
+        struct 
```

Comment:
maybe these struct could be put outside and given a type name to clean up things a bit ?

Reply:
also for the frequency we likely want a logarithmic mapping as we want more precision in the lower frequencies since our ears are more sensitive to, say, a 10hz variation around 100hz than  around 10khz.

see e.g. 

https://celtera.github.io/avendish/writing_processors/ports.metadatas.mapping.html